Ginger/GingerCoreCommon/VariablesLib/VariableNumber.cs

```diff
@@ -0,0 +1,284 @@
+                {
+                    mMaxValue = GetMaxValue();
+                }
+                return mMaxValue;             
+            }
+            set
+            { 
+                mMaxValue = value;
+                OnPropertyChanged("MaxValue");
+                OnPropertyChanged("InitialNumberValue");
+                OnPropertyChanged("Formula");
+            }
+        }
+
+        private string GetMaxValue()
+        {
+            if(!IsDecimalValue)
+            {
+                return Int32.MaxValue.ToString();
+            }
+            else
+            {
+                return float.MaxValue.ToString();
+            }
+        }
+
+        [IsSerializedForLocalRepository]
+        public string PrecisionValue
+        {
+            get
+            {
+                if(string.IsNullOrEmpty(mPrecisionValue))
+                {
+                    mPrecisionValue = "2";
+                }
+                return mPrecisionValue;
+            }
+            set
+            {
+                mPrecisionValue = value;
+                OnPropertyChanged("PrecisionValue");
+                OnPropertyChanged("InitialNumberValue");
+                OnPropertyChanged("Formula");
+            }
+        }
+
+        public override string VariableType
+        {
+            get { return "Number"; }
+        }
+
+        public override string VariableEditPage
+        {
+            get
+            {
+                return "VariableNumberPage";
+            }
+        }
+        public override eImageType Image
+        {
+            get { return eImageType.SequentialExecution; }
+        }
+
+        public override bool SupportResetValue
+        {
+            get { return true; }
+        }
+
+        public override bool SupportAutoValue
+        {
+            get { return false; }
+        }
+
+        public override bool SupportSetValue
+        {
+            get { return true; }
+        }
+
+        public override string VariableUIType
+        {
+            get { return GingerDicser.GetTermResValue(eTermResKey.Variable) + " Number"; }
+        }
+
+        [IsSerializedForLocalRepository]
+        public string InitialNumberValue
+        {
+            set
+            {
+                if(mIsDecimalValue == null)
+                {
+                    mInitialNumberValue = value;
+                }
+                else if(!Convert.ToBoolean(mIsDecimalValue))
+                {
+                    mInitialNumberValue = GetValidInteger(value);
+                }
+                else
+                {
+                    mInitialNumberValue = GetFloatWithPrecisionValue(value);
+                }
+
+                Value = mInitialNumberValue;
+                OnPropertyChanged("InitialNumberValue");
+                OnPropertyChanged("Formula");
+            }
+            get
+            {
+                if(string.IsNullOrEmpty(mInitialNumberValue))
```

Comment:
Also for here default value can be added on annotation instead of if condition here